mautic/mautic
# Tags API

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | N
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation/pull/86
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/3139
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
CRUD REST API for contact tags and small refactoring of the code around tags.

#### Steps to test this PR:
1. Since there was small refactoring, make sure tags work in the UI.
    1.a In contact form
    1.b In campaign action
    1.c In Form action
2. Test CRUD tag API and make sure the API Library tests work: https://github.com/mautic/api-library/pull/137
3. Ensure no tag duplicates were created during your tests. For example if you try to create the same tag twice.
